### PR TITLE
Use proxy authenticator

### PR DESCRIPTION
**Pull request by glatuske**
Use proxy authenticator for Ok Http client (instead of intercepting)

### CODE REVIEW

**Comment by glatuske** [comment]
See https://groups.google.com/forum/#!topic/sonarlint/2Knzpb5n0lU

File: sonar-ws/src/main/java/org/sonarqube/ws/client/OkHttpClientBuilder.java [hunk 1]
```diff
@@ -156,6 +156,12 @@ public OkHttpClient build() {
     if (readTimeoutMs >= 0) {
       builder.readTimeout(readTimeoutMs, TimeUnit.MILLISECONDS);
     }
+    if (proxyLogin != null) {
++      builder.proxyAuthenticator((route, response) -> {
```
Comment (henryju) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/OkHttpClientBuilder.java]: Merge issue?

**Comment by glatuske** [comment]
I think I should keep the Proxy-Authorization header in the interceptor.

**Comment by henryju** [comment]
@glatuske Why?

**Comment by glatuske** [comment]
Because unit tests are failing.

**Comment by henryju** [comment]
Ticket created: https://jira.sonarsource.com/browse/SONAR-9155

**Comment by henryju** [comment]
I have created a new P/R since I confirm the header is not needed: #1971 
I have also added an integration test in the scanner API: https://github.com/SonarSource/sonar-scanner-api/pull/43
I'm closing this P/R, a big thanks for reporting (and spotting) the issue.

**Pull request closed without merging**